ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateRoute/NavigateRoute.cpp

```diff
@@ -225,9 +226,12 @@ void NavigateRoute::connectRouteTrackerSignals()
       textString += "Time remaining: " + time.toString("hh:mm:ss") + "\n";
 
       // display next direction
-      if (trackingStatus->currentManeuverIndex() + 1 < dynamic_cast<DirectionManeuverListModel*>(m_directions)->directionManeuvers().length())
+      if (auto directionList = dynamic_cast<DirectionManeuverListModel*>(m_directions))
```

Comment:
Ideal solution.

Reply:
Sorry, just getting a vague recollection. Doesn't the samples team tend to avoid auto in order to be more explicit? @anmacdonald is that true?

Reply:
(Though personally I prefer it here. The type is fully visible on the right).

Reply:
We avoid `auto` in samples unless the type is an iterator or a lambada AFAIK! This is so consumers know the exact types. 

Elliot is right, it's pretty clear here what the types involved are. I'd transform `auto` into `DirectionManeuverListModel*` still, but that is just my opinion.

Reply:
Okay will do. Thanks both